src/main.rs

```diff
@@ -157,3 +226,53 @@ fn parse_line(line: &str) -> anyhow::Result<Option<DVec3>> {
         _ => Ok(None),
     }
 }
+
+fn histogram(positions: &Vec<DVec3>, r_variable: fn(&DVec3) -> f64) -> (Vec<Vec<DVec3>>, Vec<f64>) {
+    let mut position_set: Vec<DVec3> = positions.clone();
+
+    let n = position_set.len();
+    let avg = position_set.iter().copied().sum::<DVec3>() / n as f64;
+    let std_dev = (position_set
+        .iter()
+        .copied()
+        .map(|r| (r - avg).powf(2.))
+        .sum::<DVec3>()
+        / (n - 1) as f64)
+        .powf(0.5);
+
+    let cutoff: i32 = 3; // measured in standard deviations
+    let div: i32 = 6;
+    let mut range = (-(cutoff * div)..(cutoff * div))
+        .into_iter()
+        .map(|i| (i as f64) / (div as f64) * r_variable(&std_dev) + r_variable(&avg))
+        .peekable();
+    position_set.sort_by(|a, b| {
+        r_variable(a)
+            .partial_cmp(&r_variable(b))
+            .expect("Histogram: Incomparable values")
+    });
+
+    let division_values = range.clone().collect::<Vec<f64>>();
+    let mut histogram: Vec<Vec<DVec3>> = Vec::new();
+    let mut division: Vec<DVec3> = Vec::new();
+
+    for pos in position_set {
+        while range.peek().is_some() {
```

Comment:
Can't you use `while let Some(val) = range.next()` here?

Reply:
That seems a better solution (but with peek() rather than next() as value shouldn't change on every iteration of pos).

Reply:
> value shouldn't change on every iteration of pos).

Doesn't matter very much as you're either breaking or calling next on each iteration.

Reply:
What do you mean with breaking?